shared/actions/chat.js

```diff
@@ -1759,6 +1834,39 @@ function _threadIsCleared (originalAction: Action, checkAction: Action): boolean
   return originalAction.type === 'chat:loadMoreMessages' && checkAction.type === 'chat:clearMessages' && originalAction.conversationIDKey === checkAction.conversationIDKey
 }
 
+function * _getInboxAndUnbox ({payload: {conversationIDKey}}: Constants.GetInboxAndUnbox) {
+  const param: ChatTypes.localGetInboxAndUnboxLocalRpcParam = {
+    identifyBehavior: TlfKeysTLFIdentifyBehavior.chatGui,
+    query: {
+      convID: keyToConversationID(conversationIDKey),
+      computeActiveList: true,
+      tlfVisibility: CommonTLFVisibility.private,
+      topicType: CommonTopicType.chat,
+      unreadOnly: false,
+      readOnly: true,
+    },
+  }
+
+  const result: ChatTypes.GetInboxAndUnboxLocalRes = yield call(ChatTypes.localGetInboxAndUnboxLocalRpcPromise, {param})
+  const {conversations} = result
+  if (conversations && conversations[0]) {
+    yield call(_updateInbox, conversations[0])
+  }
+}
+
+function * _openConversation ({payload: {conversationIDKey}}: Constants.OpenConversation): SagaGenerator<any, any> {
+  const inbox = yield select(inboxSelector)
+  const validInbox = inbox.find(c => c.get('conversationIDKey') === conversationIDKey && c.get('validated'))
+  if (!validInbox) {
+    yield put(({type: 'chat:getInboxAndUnbox', payload: {conversationIDKey}}: Constants.GetInboxAndUnbox))
+    yield take(a => a.type === 'chat:updateInbox' && a.payload.conversation && a.payload.conversation.conversationIDKey === conversationIDKey)
```

Comment:
The coupling here seems a too loose. What happens if there's an error and a `chat:updateInbox` event isn't immediately dispatched? Is there an alternative?